File: CS/SpreadsheetExample.Module/BusinessObjects/Report.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.Xpo;

namespace SpreadsheetExample.Module.BusinessObjects {
    [NavigationItem]
    public class Report :BaseObject {
        public Report(Session session) : base(session) { }

        [RuleRequiredField]
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { 
                bool modified = SetPropertyValue("Name", value);
                if(modified && !IsLoading)
                    Data.Subject = value;
            }
        }

        [Aggregated]
        [VisibleInListView(false)]
        [ExpandObjectMembers(ExpandObjectMembers.Never)]
        public SpreadsheetDocument Data {
            get { return GetPropertyValue<SpreadsheetDocument>("Data"); }
            set { SetPropertyValue("Data", value); }
        }

        public override void AfterConstruction() {
            base.AfterConstruction();
            Data = new SpreadsheetDocument(Session);
        }
    }
}

File: VB/SpreadsheetExample.Module/BusinessObjects/Report.vb
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation
Imports DevExpress.Xpo

Namespace SpreadsheetExample.Module.BusinessObjects
	<NavigationItem>
	Public Class Report
		Inherits BaseObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<RuleRequiredField>
		Public Property Name() As String
			Get
				Return GetPropertyValue(Of String)("Name")
			End Get
			Set(ByVal value As String)
				Dim modified As Boolean = SetPropertyValue("Name", value)
				If modified AndAlso (Not IsLoading) Then
					Data.Subject = value
				End If
			End Set
		End Property

		<Aggregated, VisibleInListView(False), ExpandObjectMembers(ExpandObjectMembers.Never)>
		Public Property Data() As SpreadsheetDocument
			Get
				Return GetPropertyValue(Of SpreadsheetDocument)("Data")
			End Get
			Set(ByVal value As SpreadsheetDocument)
				SetPropertyValue("Data", value)
			End Set
		End Property

		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
			Data = New SpreadsheetDocument(Session)
		End Sub
	End Class
End Namespace
